tests/contrib/hooks/test_redis_hook.py

```diff
@@ -19,32 +19,116 @@
 
 
 import unittest
-from mock import patch
-
+from mock import patch, MagicMock
 from airflow import configuration
 from airflow.contrib.hooks.redis_hook import RedisHook
 
 
 class TestRedisHook(unittest.TestCase):
+
     def setUp(self):
         configuration.load_test_config()
 
-    def test_get_conn(self):
+    @patch('airflow.contrib.hooks.redis_hook.StrictRedis')
+    @patch('airflow.contrib.hooks.redis_hook.RedisHook.get_connection')
+    def test_get_conn(self, redis_hook_get_connection_mock, StrictRedisMock):
+        HOST = 'localhost'
+        PORT = 6379
+        PASSWORD = 's3cret!'
+        DB = 0
+
+        extra_dejson_mock = MagicMock()
+        extra_dejson_mock.get.return_value = DB
+        connection_parameters = MagicMock()
+        connection_parameters.configure_mock(
+            host=HOST,
+            port=PORT,
+            password=PASSWORD,
+            extra_dejson=extra_dejson_mock)
+        redis_hook_get_connection_mock.return_value = connection_parameters
+
+        hook = RedisHook(redis_conn_id='redis_default')
+        self.assertEqual(hook.redis, None)
+
+        self.assertEqual(hook.host, None, 'host initialised as None.')
+        self.assertEqual(hook.port, None, 'port initialised as None.')
+        self.assertEqual(hook.password, None, 'password initialised as None.')
+        self.assertEqual(hook.db, None, 'db initialised as None.')
+
+        self.assertIs(hook.get_conn(), hook.get_conn(), 'Connection initialized only if None.')
+
+        StrictRedisMock.assert_called_once_with(
+            host=HOST,
+            port=PORT,
+            password=PASSWORD,
+            db=DB)
+
+    @patch('airflow.contrib.hooks.redis_hook.StrictRedis')
+    @patch('airflow.contrib.hooks.redis_hook.RedisHook.get_connection')
```

Comment:
Since we now have a Redis running, we don't need to mock the hook anymore.

Reply:
Done.

Reply:
Done.